spec/features/admin/orders_spec.rb

```diff
@@ -201,11 +202,54 @@ def new_order_with_distribution(distributor, order_cycle)
         Spree::Config[:enable_receipt_printing?] = true
 
         distributor1.update_attribute(:abn, '12345678')
-        @order = create(:completed_order_with_totals, distributor: distributor1)
+        @order = create(:order_with_taxes, distributor: distributor1)
+        Spree::TaxRate.adjust(@order)
 
         visit spree.admin_order_path(@order)
       end
 
+      scenario "shows a list of line_items" do
+        within('table.index tbody', match: :first) do
+          @order.line_items.each do |item|
+            expect(page).to have_selector "td", match: :first, text: item.full_name
+            expect(page).to have_selector "td.price", text: item.single_display_amount
+            expect(page).to have_selector "td.qty", text: item.quantity
+            expect(page).to have_selector "td.total", text: item.display_amount
+          end
+        end
+      end
+
+      scenario "shows the order subtotal" do
+        within('table.index tbody#subtotal') do
+          expect(page).to have_selector "td.total", text: @order.display_item_total
+        end
+      end
+
+      scenario "shows the order charges (non-tax adjustments)" do
+        within('table.index tbody#order-charges') do
+          @order.adjustments.eligible.each do |adjustment|
+            next if (adjustment.originator_type == 'Spree::TaxRate') && (adjustment.amount == 0)
+            expect(page).to have_selector "td", match: :first, text: adjustment.label
+            expect(page).to have_selector "td.total", text: adjustment.display_amount
+          end
+        end
+      end
+
+      scenario "shows the order total" do
+        within('table.index tbody#order-total') do
+          expect(page).to have_selector "td.total", text: @order.display_total
+        end
+      end
+
+      scenario "shows the order taxes" do
+        within('table.index tbody#price-adjustments') do
+          @order.price_adjustment_totals.each do |label, total|
```

Comment:
I think we should rather calculate the appropriate totals without having to depend on `price_adjustment_totals`. If we happen to refactor this logic by removing this method, this spec should keep working.

Reply:
It's a feature test, we test that the view uses the method, not how the method works, right? if we did that we would simulate the method by copying all of it... or do you mean that I should use `tax_adjustment_totals` instead?

Reply:
> we test that the view uses the method

This is what views tests are for. Feature tests are just the idiomatic way to call integration tests in Rails land, I believe. As such you're executing the whole tech stack, including the browser. That's why we should not depend on a particular method and assert the behavior instead.